tests/test_utils/gcp_system_decorator.py

```diff
@@ -0,0 +1,194 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+import os
+import subprocess
+import unittest
+from contextlib import contextmanager
+from typing import Optional
+
+from tests.contrib.utils.gcp_authenticator import GcpAuthenticator
+from tests.contrib.utils.run_once_decorator import run_once
+
+AIRFLOW_MAIN_FOLDER = os.path.realpath(
+    os.path.join(
+        os.path.dirname(os.path.realpath(__file__)), os.pardir, os.pardir, os.pardir
+    )
+)
+
+AIRFLOW_PARENT_FOLDER = os.path.realpath(
+    os.path.join(AIRFLOW_MAIN_FOLDER, os.pardir, os.pardir, os.pardir)
+)
+ENV_FILE_RETRIEVER = os.path.join(
+    AIRFLOW_PARENT_FOLDER, "get_system_test_environment_variables.py"
+)
+
+AIRFLOW_HOME = os.environ.get(
+    "AIRFLOW_HOME", os.path.join(os.path.expanduser("~"), "airflow")
+)
+
+POSTGRES_LOCAL_EXECUTOR = os.path.realpath(
+    os.path.join(
+        AIRFLOW_HOME, "tests", "contrib", "operators", "postgres_local_executor.cfg"
+    )
+)
+
+SKIP_NO_GCP_CONN_ID = """
+The test is only run when AIRFLOW_CONN_GOOGLE_CLOUD_DEFAULT environment variable is set.
```

Comment:
I would prefer that the configuration of the connection be carried out in tests, because it is possible to configure the connection that some tests will be performed correctly and others will not be done correctly. It is better to avoid this.

Reply:
I think no system test configured gcp conn, so my idea was to use this simple check to skip the test instead of running test that will probably fail.